examples/siro_sandbox/gui_avatar_switch_helper.py

```diff
@@ -0,0 +1,154 @@
+        Assumes naming format between model and motion files: ("female_0.urdf" -> "female_0_motion_data_smplx.pkl")
+        """
+        models = []
+        urdfs = find_files(filepath, file_endswith, ".urdf")
+        for urdf in urdfs:
+            model_name = urdf.split(".urdf")[0]
+            motion_path = model_name + "_motion_data_smplx.pkl"
+            if os.path.isfile(motion_path):
+                models.append((urdf, motion_path))
+        return models
+
+    def _get_next_model(self):
+        model = self._humanoid_models[self._avatar_model_idx]
+        self._avatar_model_idx = (self._avatar_model_idx + 1) % len(
+            self._humanoid_models
+        )
+
+        return model
+
+    def switch_avatar(self):
+        """Switch the avatar to a new model."""
+        sim = self._sandbox_service.sim
+        config = self._sandbox_service.config
+        args = self._sandbox_service.args
+
+        model = self._get_next_model()
+
+        # record current humanoid state
+        humanoid_agent_data: ArticulatedAgentData = sim.agents_mgr[
+            self._agent_idx
+        ]
+        humanoid = humanoid_agent_data.articulated_agent
+        humanoid_controller = self._gui_agent_ctrl._humanoid_controller
+
+        prev_humanoid_base_state = (
+            humanoid.base_pos,
+            float(humanoid.base_rot),
+            humanoid_controller.prev_orientation,
+            humanoid_controller.walk_mocap_frame,
+        )
+
+        sim.get_articulated_object_manager().remove_object_by_handle(
+            humanoid.sim_obj.handle
+        )
+
+        # create new humanoid
+        humanoid = KinematicHumanoid(urdf_path=model[0], sim=sim)
+        humanoid.reconfigure()
+
+        humanoid.base_pos = prev_humanoid_base_state[0]
+        humanoid.base_rot = prev_humanoid_base_state[1]
+
+        # create new grasp manager
+        graps_mgr = RearrangeGraspManager(
+            sim, config.habitat.simulator, humanoid, 0
+        )
+
+        # maybe re-snap object
+        assert len(humanoid_agent_data.grasp_mgrs) == 1
+        if humanoid_agent_data.grasp_mgrs[0].snap_idx is not None:
+            grasped_obj_id = humanoid_agent_data.grasp_mgrs[0].snap_idx
+            graps_mgr.snap_to_obj(grasped_obj_id)
+
+        humanoid_agent_data.articulated_agent = humanoid
+        humanoid_agent_data.grasp_mgrs = [graps_mgr]
+
+        # create new humanoid controller
+        humanoid_controller = HumanoidRearrangeController(
+            walk_pose_path=model[1]
+        )
+        humanoid_controller.set_framerate_for_linspeed(
+            args.lin_speed, args.ang_speed, sim.ctrl_freq
+        )
+
+        humanoid_controller.reset(humanoid.base_transformation)
+
+        humanoid_controller.prev_orientation = prev_humanoid_base_state[2]
+        humanoid_controller.walk_mocap_frame = prev_humanoid_base_state[3]
+
+        self._gui_agent_ctrl._humanoid_controller = humanoid_controller
```

Comment:
Would it be worth doing this switching/reset logic directly in a `_gui_agent_ctrl.set_humanoid()` function?